AsyncPing.h is not found when compiling in esphome

Hi

when trying to link an esphome project in Homeassistant / ESPHome Docker it cannot find AsyncPing.h for esp32 and esp8266 ?

Is there some other link to use that ?

Greetings

Kilowatt

Hi

here is my ping.yaml including the external reference (which is working). I tried it with lokal repo too, same problem :

`external_components:
  - source:
      type: git
      url: https://github.com/trombik/esphome-component-ping
      ref: main  

#  - source:
#      type: local
#      path: "components/esphome-component-ping/components"

sensor:
  - platform: ping
    ip_address: 8.8.8.8
    num_attempts: 10
    timeout: 1sec

    loss:
      name: "${upper_devicename} Ping Packet loss" 
      id: "${devicename}_ping_packet_loss"
      
    latency:
      name: "${upper_devicename} Ping Latency"
      id: "${devicename}_ping_latency"      
      accuracy_decimals: 3`

The Error I get :

**Compiling /data/nodemcuv3/.pioenvs/nodemcuv3/src/main.cpp.o
In file included from src/esphome.h:44:0,
                 from src/main.cpp:3:
src/esphome/components/ping/ping_esp8266.h:20:23: fatal error: AsyncPing.h: No such file or directory

*******************************************************************
* Looking for AsyncPing.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:AsyncPing.h"
* Web  > https://platformio.org/lib/search?query=header:AsyncPing.h
*
*******************************************************************

 #include "AsyncPing.h"
                       ^
compilation terminated.
*** [/data/nodemcuv3/.pioenvs/nodemcuv3/src/main.cpp.o] Error 1
========================= [FAILED] Took 10.05 seconds =========================**


Thank you for helping,

Greetings from Austria

Kilowatt

i did compile with `type: git` successfully. please paste the log again. there should be missing details in the log.

```console
> esphome version 
WARNING Calling ESPHome with the configuration before the command is deprecated and will be removed in the future. 
WARNING Please instead use:
WARNING    esphome version
Version: 1.19.4
```

fyi: use ``` (triple backticks) to format log, or code.

```console
> cat ping.yaml 
substitutions:
  my_name: ping_sensor

esphome:
  name: ${my_name}
  platform: ESP8266
  board: nodemcuv2
  libraries:
    - https://github.com/akaJes/AsyncPing#95ac7e4

logger:

web_server:

ota:
  password: !secret ota_password

wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password
  ap:
    ssid: Ping
    password: !secret wifi_ap_password

captive_portal:

external_components:
  - source:
      type: git
      url: https://github.com/trombik/esphome-component-ping
      ref: main

sensor:
  - platform: ping

    # IP address of the target
    ip_address: 8.8.8.8

    # number of packets to send
    num_attempts: 17

    # the timeout. however, this is not what you usually expect from `ping`
    # implementation: the timeout is also the interval to send packets. if you
    # set this value to 10 sec, and the network is fine (no packet loss), then
    # the component sends a packet at 10 sec interval, and the total time to
    # finish would be 10 sec * num_attempts = 10 * 17 = 170 sec.
    timeout: 1sec

    loss:
      # the name to be shown.
      name: Packet loss

    latency:
      # the name to be shown.
      name: Latency
      # this should be 3 as the value is float, unit is sec, and the raw
      # values are in ms.
      accuracy_decimals: 3

```

here is the log:

```console
> esphome compile ping.yaml
INFO Reading configuration ping.yaml...
INFO Generating C++ source...
INFO Core config or version changed, cleaning build files...
INFO Compiling app...
INFO Running:  platformio run -d ping_sensor
Processing ping_sensor (board: nodemcuv2; framework: arduino; platform: platformio/espressif8266@2.6.2)
----------------------------------------------------------------------------------------------------------------------------------------------------
HARDWARE: ESP8266 80MHz, 80KB RAM, 4MB Flash
PACKAGES: 
 - framework-arduinoespressif8266 3.20704.0 (2.7.4) 
 - tool-esptool 1.409.0 (4.9) 
 - tool-esptoolpy 1.20800.0 (2.8.0) 
 - toolchain-xtensa 2.40802.200502 (4.8.2)
Library Manager: Installing ESPAsyncTCP-esphome @ 1.2.3
Library Manager: ESPAsyncTCP-esphome @ 1.2.3 has been installed!
Library Manager: Installing git+https://github.com/akaJes/AsyncPing#95ac7e4
git version 2.32.0
Cloning into '/usr/home/trombik/.platformio/.cache/tmp/pkg-installing-1lfe948h'...
HEAD is now at 95ac7e4 Fixes crash that occurs when an AsynPing object is destroyed while timers are running (#4)
Library Manager: AsyncPing(esp8266) @ 1.1.0+sha.95ac7e4 has been installed!
Library Manager: Installing ESPAsyncWebServer-esphome @ 1.2.7
Library Manager: Warning! More than one package has been found by ESPAsyncWebServer-esphome @ 1.2.7 requirements:
 - ottowinter/ESPAsyncWebServer-esphome @ 2.0.1
 - esphome/ESPAsyncWebServer-esphome @ 2.0.1
Library Manager: Please specify detailed REQUIREMENTS using package owner and version (showed above) to avoid name conflicts
Library Manager: ESPAsyncWebServer-esphome @ 1.2.7 has been installed!
Library Manager: Installing dependencies...
Library Manager: Installing AsyncTCP-esphome
Library Manager: Warning! More than one package has been found by AsyncTCP-esphome requirements:
 - ottowinter/AsyncTCP-esphome @ 1.2.1
 - esphome/AsyncTCP-esphome @ 1.2.2
Library Manager: Please specify detailed REQUIREMENTS using package owner and version (showed above) to avoid name conflicts
Library Manager: AsyncTCP-esphome @ 1.2.1 has been installed!
Library Manager: Installing Hash
Library Manager: Already installed, built-in library
Library Manager: Installing Update
Library Manager: Already installed, built-in library
Library Manager: Installing ArduinoJson-esphomelib @ 5.13.3
Library Manager: ArduinoJson-esphomelib @ 5.13.3 has been installed!
Dependency Graph
|-- <ESPAsyncTCP-esphome> 1.2.3
|   |-- <ESP8266WiFi> 1.0
|-- <AsyncPing(esp8266)> 1.1.0+sha.95ac7e4
|   |-- <ESP8266WiFi> 1.0
|-- <ESPAsyncWebServer-esphome> 1.2.7
|   |-- <ESPAsyncTCP-esphome> 1.2.3
|   |   |-- <ESP8266WiFi> 1.0
|   |-- <Hash> 1.0
|   |-- <ESP8266WiFi> 1.0
|   |-- <ArduinoJson-esphomelib> 5.13.3
|-- <ESP8266WiFi> 1.0
|-- <ESP8266mDNS> 1.2
|   |-- <ESP8266WiFi> 1.0
|-- <ArduinoJson-esphomelib> 5.13.3
|-- <DNSServer> 1.1.1
|   |-- <ESP8266WiFi> 1.0
Compiling .pioenvs/ping_sensor/src/esphome/components/captive_portal/captive_portal.cpp.o
Compiling .pioenvs/ping_sensor/src/esphome/components/json/json_util.cpp.o
Compiling .pioenvs/ping_sensor/src/esphome/components/logger/logger.cpp.o
Compiling .pioenvs/ping_sensor/src/esphome/components/ota/ota_component.cpp.o
Compiling .pioenvs/ping_sensor/src/esphome/components/ping/ping_sock.c.o
Compiling .pioenvs/ping_sensor/src/esphome/components/sensor/automation.cpp.o
Compiling .pioenvs/ping_sensor/src/esphome/components/sensor/filter.cpp.o
Compiling .pioenvs/ping_sensor/src/esphome/components/sensor/sensor.cpp.o
Compiling .pioenvs/ping_sensor/src/esphome/components/web_server/web_server.cpp.o
Compiling .pioenvs/ping_sensor/src/esphome/components/web_server_base/web_server_base.cpp.o
Compiling .pioenvs/ping_sensor/src/esphome/components/wifi/wifi_component.cpp.o
Compiling .pioenvs/ping_sensor/src/esphome/components/wifi/wifi_component_esp32.cpp.o
Compiling .pioenvs/ping_sensor/src/esphome/components/wifi/wifi_component_esp8266.cpp.o
Compiling .pioenvs/ping_sensor/src/esphome/core/application.cpp.o
Compiling .pioenvs/ping_sensor/src/esphome/core/component.cpp.o
Compiling .pioenvs/ping_sensor/src/esphome/core/controller.cpp.o
Compiling .pioenvs/ping_sensor/src/esphome/core/esphal.cpp.o
Compiling .pioenvs/ping_sensor/src/esphome/core/helpers.cpp.o
Compiling .pioenvs/ping_sensor/src/esphome/core/log.cpp.o
Compiling .pioenvs/ping_sensor/src/esphome/core/preferences.cpp.o
Compiling .pioenvs/ping_sensor/src/esphome/core/scheduler.cpp.o
Compiling .pioenvs/ping_sensor/src/esphome/core/util.cpp.o
Compiling .pioenvs/ping_sensor/src/main.cpp.o
Generating LD script .pioenvs/ping_sensor/ld/local.eagle.app.v6.common.ld
Compiling .pioenvs/ping_sensor/lib3b8/ESP8266WiFi/BearSSLHelpers.cpp.o
Compiling .pioenvs/ping_sensor/lib3b8/ESP8266WiFi/CertStoreBearSSL.cpp.o
Compiling .pioenvs/ping_sensor/lib3b8/ESP8266WiFi/ESP8266WiFi.cpp.o
Compiling .pioenvs/ping_sensor/lib3b8/ESP8266WiFi/ESP8266WiFiAP.cpp.o
Compiling .pioenvs/ping_sensor/lib3b8/ESP8266WiFi/ESP8266WiFiGeneric.cpp.o
Compiling .pioenvs/ping_sensor/lib3b8/ESP8266WiFi/ESP8266WiFiGratuitous.cpp.o
Compiling .pioenvs/ping_sensor/lib3b8/ESP8266WiFi/ESP8266WiFiMulti.cpp.o
Compiling .pioenvs/ping_sensor/lib3b8/ESP8266WiFi/ESP8266WiFiSTA-WPS.cpp.o
Compiling .pioenvs/ping_sensor/lib3b8/ESP8266WiFi/ESP8266WiFiSTA.cpp.o
Compiling .pioenvs/ping_sensor/lib3b8/ESP8266WiFi/ESP8266WiFiScan.cpp.o
Compiling .pioenvs/ping_sensor/lib3b8/ESP8266WiFi/WiFiClient.cpp.o
Compiling .pioenvs/ping_sensor/lib3b8/ESP8266WiFi/WiFiClientSecureAxTLS.cpp.o
Compiling .pioenvs/ping_sensor/lib3b8/ESP8266WiFi/WiFiClientSecureBearSSL.cpp.o
Compiling .pioenvs/ping_sensor/lib3b8/ESP8266WiFi/WiFiServer.cpp.o
Compiling .pioenvs/ping_sensor/lib3b8/ESP8266WiFi/WiFiServerSecureAxTLS.cpp.o
Compiling .pioenvs/ping_sensor/lib3b8/ESP8266WiFi/WiFiServerSecureBearSSL.cpp.o
Compiling .pioenvs/ping_sensor/lib3b8/ESP8266WiFi/WiFiUdp.cpp.o
Compiling .pioenvs/ping_sensor/lib88f/ESPAsyncTCP-esphome/AsyncPrinter.cpp.o
Compiling .pioenvs/ping_sensor/lib88f/ESPAsyncTCP-esphome/ESPAsyncTCP.cpp.o
Compiling .pioenvs/ping_sensor/lib88f/ESPAsyncTCP-esphome/ESPAsyncTCPbuffer.cpp.o
Compiling .pioenvs/ping_sensor/lib88f/ESPAsyncTCP-esphome/SyncClient.cpp.o
Compiling .pioenvs/ping_sensor/lib88f/ESPAsyncTCP-esphome/tcp_axtls.c.o
Compiling .pioenvs/ping_sensor/lib8df/AsyncPing_esp8266_/AsyncPing.cpp.o
Compiling .pioenvs/ping_sensor/lib0f3/Hash/Hash.cpp.o
Archiving .pioenvs/ping_sensor/lib84d/libArduinoJson-esphomelib.a
Indexing .pioenvs/ping_sensor/lib84d/libArduinoJson-esphomelib.a
Compiling .pioenvs/ping_sensor/lib00e/ESPAsyncWebServer-esphome/AsyncEventSource.cpp.o
Compiling .pioenvs/ping_sensor/lib00e/ESPAsyncWebServer-esphome/AsyncWebSocket.cpp.o
Archiving .pioenvs/ping_sensor/lib0f3/libHash.a
Indexing .pioenvs/ping_sensor/lib0f3/libHash.a
Compiling .pioenvs/ping_sensor/lib00e/ESPAsyncWebServer-esphome/SPIFFSEditor.cpp.o
Compiling .pioenvs/ping_sensor/lib00e/ESPAsyncWebServer-esphome/WebAuthentication.cpp.o
Compiling .pioenvs/ping_sensor/lib00e/ESPAsyncWebServer-esphome/WebHandlers.cpp.o
Archiving .pioenvs/ping_sensor/lib3b8/libESP8266WiFi.a
Indexing .pioenvs/ping_sensor/lib3b8/libESP8266WiFi.a
Archiving .pioenvs/ping_sensor/lib8df/libAsyncPing_esp8266_.a
Indexing .pioenvs/ping_sensor/lib8df/libAsyncPing_esp8266_.a
Archiving .pioenvs/ping_sensor/lib88f/libESPAsyncTCP-esphome.a
Indexing .pioenvs/ping_sensor/lib88f/libESPAsyncTCP-esphome.a
Compiling .pioenvs/ping_sensor/lib00e/ESPAsyncWebServer-esphome/WebRequest.cpp.o
Compiling .pioenvs/ping_sensor/lib00e/ESPAsyncWebServer-esphome/WebResponses.cpp.o
Compiling .pioenvs/ping_sensor/lib00e/ESPAsyncWebServer-esphome/WebServer.cpp.o
In file included from .piolibdeps/ping_sensor/ESPAsyncWebServer-esphome/src/SPIFFSEditor.cpp:1:0:
.piolibdeps/ping_sensor/ESPAsyncWebServer-esphome/src/SPIFFSEditor.h:16:101: warning: 'SPIFFS' is deprecated (declared at /usr/home/trombik/.platformio/packages/framework-arduinoespressif8266@3.20704.0/cores/esp8266/FS.h:269): SPIFFS has been deprecated. Please consider moving to LittleFS or other filesystems. [-Wdeprecated-declarations]
     SPIFFSEditor(const String& username=String(), const String& password=String(), const fs::FS& fs=SPIFFS);
                                                                                                     ^
Compiling .pioenvs/ping_sensor/libfdb/ESP8266mDNS/ESP8266mDNS.cpp.o
Compiling .pioenvs/ping_sensor/libfdb/ESP8266mDNS/ESP8266mDNS_Legacy.cpp.o
Compiling .pioenvs/ping_sensor/libfdb/ESP8266mDNS/LEAmDNS.cpp.o
Compiling .pioenvs/ping_sensor/libfdb/ESP8266mDNS/LEAmDNS_Control.cpp.o
Compiling .pioenvs/ping_sensor/libfdb/ESP8266mDNS/LEAmDNS_Helpers.cpp.o
Compiling .pioenvs/ping_sensor/libfdb/ESP8266mDNS/LEAmDNS_Structs.cpp.o
Compiling .pioenvs/ping_sensor/libfdb/ESP8266mDNS/LEAmDNS_Transfer.cpp.o
Compiling .pioenvs/ping_sensor/lib24b/DNSServer/DNSServer.cpp.o
Archiving .pioenvs/ping_sensor/lib00e/libESPAsyncWebServer-esphome.a
Indexing .pioenvs/ping_sensor/lib00e/libESPAsyncWebServer-esphome.a
Archiving .pioenvs/ping_sensor/libFrameworkArduinoVariant.a
Indexing .pioenvs/ping_sensor/libFrameworkArduinoVariant.a
Compiling .pioenvs/ping_sensor/FrameworkArduino/Crypto.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/Esp-frag.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/Esp-version.cpp.o
Archiving .pioenvs/ping_sensor/lib24b/libDNSServer.a
Indexing .pioenvs/ping_sensor/lib24b/libDNSServer.a
Compiling .pioenvs/ping_sensor/FrameworkArduino/Esp.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/FS.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/FSnoop.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/FunctionalInterrupt.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/HardwareSerial.cpp.o
Archiving .pioenvs/ping_sensor/libfdb/libESP8266mDNS.a
Indexing .pioenvs/ping_sensor/libfdb/libESP8266mDNS.a
Compiling .pioenvs/ping_sensor/FrameworkArduino/IPAddress.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/MD5Builder.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/Print.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/Schedule.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/StackThunk.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/Stream.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/StreamString.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/Tone.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/TypeConversion.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/Updater.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/WMath.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/WString.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/abi.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/base64.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/cbuf.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/cont.S.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/cont_util.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/core_esp8266_app_entry_noextra4k.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/core_esp8266_eboot_command.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/core_esp8266_features.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/core_esp8266_flash_quirks.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/core_esp8266_flash_utils.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/core_esp8266_i2s.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/core_esp8266_main.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/core_esp8266_noniso.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/core_esp8266_phy.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/core_esp8266_postmortem.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/core_esp8266_si2c.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/core_esp8266_sigma_delta.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/core_esp8266_spi_utils.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/core_esp8266_timer.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/core_esp8266_waveform.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/core_esp8266_wiring.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/core_esp8266_wiring_analog.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/core_esp8266_wiring_digital.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/core_esp8266_wiring_pulse.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/core_esp8266_wiring_pwm.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/core_esp8266_wiring_shift.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/crc32.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/debug.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/flash_hal.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/gdb_hooks.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/heap.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/libb64/cdecode.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/libb64/cencode.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/libc_replacements.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/sntp-lwip2.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/spiffs/spiffs_cache.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/spiffs/spiffs_check.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/spiffs/spiffs_gc.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/spiffs/spiffs_hydrogen.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/spiffs/spiffs_nucleus.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/spiffs_api.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/sqrt32.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/time.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/uart.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/umm_malloc/umm_info.c.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/umm_malloc/umm_integrity.c.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/umm_malloc/umm_local.c.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/umm_malloc/umm_malloc.cpp.o
Compiling .pioenvs/ping_sensor/FrameworkArduino/umm_malloc/umm_poison.c.o
Archiving .pioenvs/ping_sensor/libFrameworkArduino.a
Indexing .pioenvs/ping_sensor/libFrameworkArduino.a
Linking .pioenvs/ping_sensor/firmware.elf
Retrieving maximum program size .pioenvs/ping_sensor/firmware.elf
Building .pioenvs/ping_sensor/firmware.bin
Checking size .pioenvs/ping_sensor/firmware.elf
RAM:   [====      ]  39.8% (used 32628 bytes from 81920 bytes)
Flash: [====      ]  37.8% (used 395216 bytes from 1044464 bytes)
Creating BIN file ".pioenvs/ping_sensor/firmware.bin" using "/usr/home/trombik/.platformio/packages/framework-arduinoespressif8266@3.20704.0/bootloaders/eboot/eboot.elf" and ".pioenvs/ping_sensor/firmware.elf"
=========================================================== [SUCCESS] Took 31.18 seconds ===========================================================

****************************************************************************************************************************************************
There is a new version 5.2.2 of PlatformIO available.
Please upgrade it via `platformio upgrade` or `pip install -U platformio` command.
Changes: https://docs.platformio.org/en/latest/history.html
****************************************************************************************************************************************************


****************************************************************************************************************************************************
There are the new updates for platforms (espressif32, espressif8266)
Please update them via `platformio platform update` command.

If you want to manually check for the new versions without updating, please use `platformio platform update --dry-run` command.
****************************************************************************************************************************************************

INFO Successfully compiled program.
```

--- a/README.md
+++ b/README.md
@@ -71,6 +71,19 @@ esphome compile config/ping.yaml
 
 See [External Components](https://esphome.io/components/external_components.html).
 
+Note that, for `esp8266`, you need to include a library
+[`AsyncPing`](https: //github.com/akaJes/AsyncPing). Make sure your config
+includes it by `libraries`.
+
+```yaml
+esphome:
+  name: ${my_name}
+  platform: ESP8266
+  board: nodemcuv2
+  libraries:
+    - https://github.com/akaJes/AsyncPing#95ac7e4
+```
+
 ### With `local`
 
 Create `components` directory in your project root.

